enzyme/Enzyme/MLIR/Passes/EnzymeBatchDiffPass.cpp

```diff
@@ -36,37 +42,272 @@ namespace mlir {
 namespace enzyme {
 namespace batchutils {
 
-bool isReadOnly(Operation *op) {
-  // If the op has memory effects, try to characterize them to see if the op
-  // is trivially dead here.
-  if (auto effectInterface = dyn_cast<MemoryEffectOpInterface>(op)) {
-    // Check to see if this op either has no effects, or only allocates/reads
-    // memory.
-    SmallVector<MemoryEffects::EffectInstance, 1> effects;
-    effectInterface.getEffects(effects);
-    if (!llvm::all_of(effects, [op](const MemoryEffects::EffectInstance &it) {
-          return isa<MemoryEffects::Read>(it.getEffect());
-        })) {
-      return false;
+Value tensorizeArg(OpBuilder &builder, Location &loc,
+                   SmallVector<Value> &argList) {
+  auto argTy = argList.front().getType();
+  auto T = dyn_cast<TensorType>(argTy);
+  mlir::Value out;
```

Comment:
this is going to break in Enzyme-JaX.

Can you instead make a new enzyme op for these operations, and then a separate lowering pass. Here we will add a lowering pass to tensor. In Enzyme-Jax we will add a lowering pass to stablehlo